### DISCUSSION TITLE
Categorizing the datasets

### DISCUSSION THREAD

**Discussion by RaulFD-creator** [polaris-hub/polaris#143] [General]
**Problem**
Currently, the Polaris website shows all datasets at once to the user. It may be helpful to categorize them either by ML task (single instance or multi-instance; classification or regression) or by biochemical task (ADME, Toxicity, Quantum properties, binding affinity, etc.).

**Posible solutions**
There are multiple questions for such a categorisation. It could be dynamic (each users decides how their dataset should be categorised) or static (there a fixed number of categories and the user uploading can decide which one best fits their dataset). The idea I think would offer the most flexibility is have two trees of categories (ML task and Biochemical task) so that users with different interests can explore in a more personalised way. 

**Discussion**
@cwognum suggested creating this discussion to get a better sense of which of the solutions might be the most attractive to the community, so feel free to add your thoughts below; it will also be interesting to see which categories would be the most interesting for everyone, so if you have any suggestions, do put them forward.

**Comment by cwognum** [comment]
Thank you @RaulFD-creator ! I think this is a great point to raise. More generally, I would be open to hear ideas on how we could improve content discovery on the Hub. This can be as simple as UI/UX improvements or as complex as LLM-powered search.

For example: 
- Maybe a user has been trained in ML, but has no experience in drug discovery yet and is trying to simply get some hands-on experience. Such a user may not know what to search for. Highlighted datasets (e.g. "Latest Arrivals" or "Most Popular") would be useful for such a user.
- Maybe a user has worked with a specific dataset or benchmark, and is curious to find similar datasets and benchmarks. Showing related artifacts on a dataset's or benchmark's page could be useful for such a user!
- Beyond categorization, a more flexible feature that lets users group similar datasets or benchmarks in collections could be useful.

One thing worth highlighting is that we ultimately aspire to support any modality or task relevant in drug discovery, not just small-molecule ones.

**Reply by RaulFD-creator** [reply]
I like the idea of that versatility and I think that the most popular feature could be really interesting. Conversely, another useful tag could be "Hardest baseline" (meaning datasets for which the leaderboard hasn't changed in a long time), thus encouraging people to look at more complex problems.

Regarding UI/UX, I think maybe visually separated windows depending on the dataset tag, so that it looks a bit more organised. If a dataset has more than one tag, you could group it with the most or least popular tag.

Regarding the idea of looking at similar datasets of benchmarks to recommend to users, you could store which datasets are downloaded by which users and create a network of co-use. I am thinking that a user A with an interest in ADME will tend to use certain datasets more than a user B with an interest in molecular representation learning who will gravitate more towards common benchmarks in the field.
